src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/Hop.tsx

```diff
@@ -201,15 +202,25 @@ export const Hop = ({
               amountCryptoBaseUnit={tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit}
             />
           )}
+
+          <Collapse in={isAllowanceResetInitiallyNeeded} style={collapseWidth}>
+            {isAllowanceResetInitiallyNeeded && (
+              <ApprovalStep
+                tradeQuoteStep={tradeQuoteStep}
+                hopIndex={hopIndex}
+                isActive={hopExecutionState === HopExecutionState.AwaitingApprovalReset}
+                isAllowanceResetStep={true}
+              />
+            )}
+          </Collapse>
           <Collapse in={isApprovalInitiallyNeeded} style={collapseWidth}>
             {isApprovalInitiallyNeeded === true && (
               <ApprovalStep
                 tradeQuoteStep={tradeQuoteStep}
                 hopIndex={hopIndex}
-                isActive={
-                  hopExecutionState === HopExecutionState.AwaitingApprovalReset ||
-                  hopExecutionState === HopExecutionState.AwaitingApproval
-                }
+                isActive={hopExecutionState === HopExecutionState.AwaitingApproval}
+                isAllowanceResetStep={false}
+                isAwaitingReset={hopExecutionState === HopExecutionState.AwaitingApprovalReset}
```

Comment:
What's this, you ask? Well, after many hours of debugging it turns out that it costs about 40% more gas to go from an allowance of `0` to an allowance of _something_, than an allowance of _something_ to _something else_.

Thus, we need to know if we are awaiting an allowance reset, so that we can recompute the allowance gas costs when it's done (if we use the value on first render it'll be too low, as it doesn't know that we are about to reset it to `0`).